src/validation.cpp

```diff
@@ -2060,6 +2060,11 @@ void Chainstate::InvalidChainFound(CBlockIndex* pindexNew)
         m_chainman.m_best_invalid = pindexNew;
     }
     if (m_chainman.m_best_header != nullptr && m_chainman.m_best_header->GetAncestor(pindexNew->nHeight) == pindexNew) {
+        CBlockIndex* index_walk{m_chainman.m_best_header};
+        while (index_walk != pindexNew) {
+            index_walk->nStatus |= BLOCK_FAILED_CHILD;
```

Comment:
I think that this is not a complete fix. This would only mark indexes between `m_best_header` and the failed block as `BLOCK_FAILED_CHILD`, but if this was not  a linear chain and there were forked blocks, those wouldn't be marked as `BLOCK_FAILED_CHILD`. I think I'll change the logic to also set `BLOCK_FAILED_CHILD` while iterating over the entire block index.

Reply:
I've reworked this by introducing `SetBlockFailureFlags` (as a counterpart to `ResetBlockFailureFlags` that removes the flags after reconsiderblock)